qiskit/transpiler/passes/optimization/inverse_cancellation.py

```diff
@@ -76,11 +80,13 @@ def run(self, dag: DAGCircuit):
         Returns:
             DAGCircuit: Transformed DAG.
         """
+        if self.self_inverse_gates:
+            dag = self._run_on_self_inverse(dag)
+        if self.inverse_gate_pairs:
+            dag = self._run_on_inverse_pairs(dag)
+        return dag
 
-        dag = self._run_on_self_inverse(dag, self.self_inverse_gates)
-        return self._run_on_inverse_pairs(dag, self.inverse_gate_pairs)
```

Comment:
Given the immediate short-circuiting within the two functions, I doubt much is being bought by these outer `if` statements, but that's not important.